hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDTableServiceClient.java

```diff
@@ -245,6 +246,7 @@ private void completeClustering(HoodieReplaceCommitMetadata metadata,
           metrics.updateCommitMetrics(parsedInstant.getTime(), durationInMs, metadata, HoodieActiveTimeline.REPLACE_COMMIT_ACTION)
       );
     }
+    waitForAsyncServiceCompletion();
     LOG.info("Clustering successfully on commit " + clusteringCommitTime);
```

Comment:
Why we need this change?

Reply:
Without this change，if config ASYNC_CLEAN = true，AsyncCleanerService will be used to do clean in offline job 。In my unit testing for offline job，if the completion time of the compact/cluster job is earlier than the completion time of the sync-cleaning job, function BaseHoodieTableServiceClient.close() will force the asynchronous  cleaning job to be closed， it will causes interrupt Excpetion.

Reply:
This is not correct, we need to wait for the async table service to finish.